resolwe_bio/processes/plots/dexseq_plot.yml

```diff
@@ -0,0 +1,75 @@
+# =============
+# Plot - DEXSeq
+# =============
+---
+
+- slug: plot-dexeq
+  name: DEXSeq - Plot
+  requirements:
+    expression-engine: jinja
+    executor:
+      docker:
+        image: resolwebio/rnaseq:2.1.1
+  data_name: 'Plot - DEXSeq (case vs. control)'
+  version: 0.0.1
+  type: data:plot:dexseq
+  category: analyses
+  persistence: CACHED
+  description: >
+    Visualise DEXSeq results.
+  input:
+    - name: dexseq
+      label: DEXSeq data object
+      type: data:differentialexpression:dexseq
+    - name: gene
+      label: Gene
+      type: basic:string
+    - name: transcript
+      label: Visualize transcript models
+      type: basic:boolean
+      default: False
+      description: >
+        Include the transcript model in the plot.
+    - name: sample
+      label: Plot individual samples
+      type: basic:boolean
+      default: False
+      description: >
+        Plot expression values for individual samples.
+    - name: exon
+      label: Plot exon expression
+      type: basic:boolean
+      default: False
+      description: >
+        Plot changes in the expression of individual exons.
+  output:
+    - name: pdf 
+      label: DEXSeq visualisation
+      type: basic:file
+  run:
+    runtime: polyglot
+    language: bash
+    program: |
+      {% if sample and exon %}
+        re-error "Either 'Plot individual samples' or 'Plot exon expression' may be chosen, not both."
+      {% endif %}
+
+      FLAGS=""
+      {% if transcript %}
+        FLAGS+=" -t"
+      {% endif %}
+
+      {% if sample %}
+        FLAGS+=" -s"
+      {% elif exon %}
+        FLAGS+=" -e"
+      {% endif %}
+
+      plot_dexseq.R \
+        --dxr {{ dexseq.dxr.file }} \
+        --gene {{ gene }} \
+        ${FLAGS}
+
+      re-checkrc "Error plotting differential expression data (DEXSeq)."
+
+      re-save-file pdf Rplots.pdf
```

Comment:
This process was envisioned to be used in visualizations and not as a stand-alone tool. That's why I didn't put too much effort in the name conventions and wanted to keep it short and not add extra lines to the process (like renaming the file) or the R script. But of course it could be done.

Reply:
I suggest to make this into `dexseq-plot-<gene_id>.pdf`. This kind of name is far better for the visualization export functionality. The analysis works only with ENSEMBL-based annotations, so there is no worry about the possible special characters e.g. `:` in the file name.